plugins/inputs/googlecoreiot/googlecoreiot.go

```diff
@@ -0,0 +1,436 @@
+package googlecoreiot
+
+import (
+	"compress/gzip"
+	"crypto/subtle"
+	"crypto/tls"
+	"encoding/base64"
+	"encoding/json"
+	"log"
+	"net"
+	"net/http"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	tlsint "github.com/influxdata/telegraf/internal/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/plugins/parsers"
+	influx "github.com/influxdata/telegraf/plugins/parsers/influx"
+)
+
+// Attributes maps Default fields sent by Google Pub/Sub
+type Attributes struct {
+	DeviceID               string `json:"deviceId"`
+	DeviceNumID            string `json:"deviceNumId"`
+	DeviceRegistryID       string `json:"deviceRegistryId"`
+	DeviceRegistryLocation string `json:"deviceRegistryLocation"`
+	ProjectID              string `json:"projectId"`
+	SubFolder              string `json:"subFolder"`
+}
+
+// Message Structure of a Google Pub/Sub message
+type Message struct {
+	Atts         map[string]interface{} `json:"attributes"`
+	Data         string                 `json:"data"`
+	MessageID    string                 `json:"messageId"`
+	MessageID2   string                 `json:"message_id"`
+	PublishTime  string                 `json:"publishTime"`
+	PublishTime2 string                 `json:"publish_time"`
+}
+
+// Payload of Line-Protocol payload after Base64-decode
+type Payload struct {
+	Msg          Message `json:"message"`
+	Subscription string  `json:"subscription"`
+}
+
+// JSONData structure of the Base-64 encoded payload
+type JSONData struct {
+	Name   string                 `json:"measurement"`
+	Tags   map[string]string      `json:"tags"`
+	Fields map[string]interface{} `json:"fields"`
+	Time   int64                  `json:"time"`
+}
+
+// DEFAULT_MAX_BODY_SIZE is the default maximum request body size, in bytes.
+// if the request body is over this size, we will return an HTTP 413 error.
+// 500 MB
+const defaultMaxBodySize = 500 * 1024 * 1024
+
+type TimeFunc func() time.Time
+
+type GoogleListener struct {
+	ServiceAddress  string
+	Path            string
+	Methods         []string
+	ReadTimeout     internal.Duration
+	WriteTimeout    internal.Duration
+	MaxBodySize     internal.Size
+	Port            int
+	Precision       string
+	DataFormat      string
+	MeasurementName string
+	handler         *influx.MetricHandler
+
+	tlsint.ServerConfig
+
+	BasicUsername string
+	BasicPassword string
```

Comment:
Remove any options no longer used, such as Methods, BasicUsername, BasicPassword.  Make sure any exported variable is documented.